virtual_rainforest/models/plants/functions.py

```diff
@@ -0,0 +1,170 @@
+"""The :mod:`~virtual_rainforest.models.plants.functions` submodule provides the core
+functions used to estimate the canopy model, productivity and growth.
+
+NOTE - much of this will be outsourced to pyrealm.
+
+"""  # noqa: D205, D415
+
+from __future__ import annotations
+
+import numpy as np
+from numpy.typing import NDArray
+from xarray import DataArray
+
+from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.exceptions import ConfigurationError, InitialisationError
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.utils import set_layer_roles
+from virtual_rainforest.models.plants.community import PlantCohort, PlantCommunities
+
+
+def generate_canopy_model(community: list[PlantCohort]) -> tuple[NDArray, NDArray]:
+    """Generate the canopy structure for a plant community.
+
+    This function takes a list of plant cohorts present in a community and uses the T
+    Model to estimate the heights and crown areas of the individuals. It then uses the
+    perfect plasticity approximation to calculate the closure heights of the canopy
+    layers and the leaf area indices of each layer.
+
+    Args:
+        community: A list of plant cohorts.
+
+    Returns:
+        A tuple of one dimensional numpy arrays giving the canopy layer heights and leaf
+        area indices.
+    """
+
+    # TODO - actually calculate these
+    layer_hght = np.array([30.0, 20.0, 10.0])
```

Comment:
Feel like `layer_height` is quite a lot clearer as a name for minimal extra length